File: code/generate_strategies/Direct/java/function_dask_bag_fold.java
package com.example;

import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
import java.util.Arrays;

public class CumulativeSumFunction {

    public String cumulative_sum(String inputStr) {
        try {
            double[] inputList = Arrays.stream(inputStr.split(","))
                    .mapToDouble(Double::parseDouble)
                    .toArray();

            DescriptiveStatistics stats = new DescriptiveStatistics();
            Arrays.stream(inputList).forEach(stats::addValue);

            double result = stats.getSum();

            return String.valueOf(result);
        } catch (Exception e) {
            return "Error";
        }
    }
}

File: manual/data/python/src/function_dask_bag_fold.py
from dask import bag

class CumulativeSumFunction:
    def cumulative_sum(self, input_str: str) -> str:
        try:
            input_list = list(map(float, input_str.split(",")))
            dask_bag = bag.from_sequence(input_list)
            result = dask_bag.fold(lambda x, y: x + y).compute()
            return str(result)
        except Exception as e:
            return "Error"
